### PR TITLE
Use timeout from dcos config for diagnostic download

### PR DESCRIPTION
**Pull request by janisz**
Before `core.timeout` was ignored now it is taken into account

```
➜ dcos config set core.timeout 3                                     
➜ time dcos diagnostics download         
Error: Get "https://master.testing.mesosphe.re/system/health/v1/diagnostics/d4d84ac3-64c7-4134-8fbc-3b2c382620f1/file": context deadline exceeded
dcos diagnostics download  0,13s user 0,03s system 4% cpu 3,798 total
➜ time dcos diagnostics download
➜ dcos config set core.timeout 30
➜ time dcos diagnostics download 
Error: context deadline exceeded
dcos diagnostics download  1,06s user 2,40s system 10% cpu 31,782 total
➜ config set core.timeout 300
➜  dcos diagnostics download  
dcos diagnostics download  6,27s user 13,87s system 21% cpu 1:33,78 total
```

Fixes: https://jira.d2iq.com/browse/COPS-6584

### CODE REVIEW

File: pkg/cmd/diagnostics/diagnostics_download.go [hunk 1]
```diff
@@ -18,7 +18,14 @@ func newDiagnosticsDownloadCommand() *cobra.Command {
 		Short: "Download diagnostics bundle",
 		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			c := pluginutil.HTTPClient("")
+			cluster, err := ctx.Cluster()
+			if err != nil {
+				return fmt.Errorf("could not get cluster: %s", err)
```
Comment (pierrebeitz) [inline pkg/cmd/diagnostics/diagnostics_download.go]: nice. all other places went with `return nil, err`, `return 0, err` or `return "", err`. 
i like a error message to grep for! 👍

**Review by pierrebeitz — APPROVED** [review]
makes sense to me

File: pkg/cmd/diagnostics/diagnostics_download.go (hunk 1 above)
Comment (janisz) [reply pkg/cmd/diagnostics/diagnostics_download.go]: It's good practice to add context to error and never return raw `err`

**Comment by janisz** [comment]
@mesosphere-ci run mac integration tests

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Pull request merged**